### PR TITLE
[Test]Refactor test/nn/test_dropout.py to be device-agnostic

### PR DESCRIPTION
**Pull request by kumartanmay-28**
Moves test_native_dropout_corner_case from TestDropoutNN to TestDropoutNNDeviceType to enable device-generic testing across all accelerators.

  Changes:
  - Moved test_native_dropout_corner_case to TestDropoutNNDeviceType class
  - Added device parameter and runtime skip for CPU variant (test compares accelerator vs CPU)
  - Removed hardcoded TEST_CUDA/TEST_PRIVATEUSE1 device selection logic
  - Extended bfloat16 support to include XPU
  - Removed unused TEST_CUDA and TEST_PRIVATEUSE1 imports

Test runs on CUDA, XPU, and other accelerators; skips on CPU variant.

Tets plan:
(cuda_torch_build) [root@88fde2b2e6e0 pytorch]# TEST_CONFIG=cuda python test/nn/test_dropout.py -v
W0518 06:31:32.232000 1131 site-packages/torch/_native/cutedsl_utils.py:56] CuTeDSL operators require optional Python packages `nvidia-cutlass-dsl` and `apache-tvm-ffi`; missing optional dependency `nvidia_cutlass_dsl` (importlib.util.find_spec(nvidia_cutlass_dsl) failed)
test_AlphaDropout (__main__.TestDropoutNN.test_AlphaDropout) ... ok
test_FeatureAlphaDropout (__main__.TestDropoutNN.test_FeatureAlphaDropout) ... ok
test_invalid_dropout_p (__main__.TestDropoutNN.test_invalid_dropout_p) ... ok
test_Dropout1d_cpu_float64 (__main__.TestDropoutNNDeviceTypeCPU.test_Dropout1d_cpu_float64) ... ok
test_Dropout2d_cpu (__main__.TestDropoutNNDeviceTypeCPU.test_Dropout2d_cpu) ... ok
test_Dropout3d_cpu (__main__.TestDropoutNNDeviceTypeCPU.test_Dropout3d_cpu) ... ok
test_Dropout_cpu (__main__.TestDropoutNNDeviceTypeCPU.test_Dropout_cpu) ... ok
test_empty_dropout_cpu (__main__.TestDropoutNNDeviceTypeCPU.test_empty_dropout_cpu) ... ok
test_native_dropout_corner_case_cpu (__main__.TestDropoutNNDeviceTypeCPU.test_native_dropout_corner_case_cpu) ... skipped 'Test requires accelerator to compare with CPU'
test_Dropout1d_cuda_float64 (__main__.TestDropoutNNDeviceTypeCUDA.test_Dropout1d_cuda_float64) ... ok
test_Dropout2d_cuda (__main__.TestDropoutNNDeviceTypeCUDA.test_Dropout2d_cuda) ... ok
test_Dropout3d_cuda (__main__.TestDropoutNNDeviceTypeCUDA.test_Dropout3d_cuda) ... ok
test_Dropout_cuda (__main__.TestDropoutNNDeviceTypeCUDA.test_Dropout_cuda) ... ok
test_empty_dropout_cuda (__main__.TestDropoutNNDeviceTypeCUDA.test_empty_dropout_cuda) ... ok
test_native_dropout_corner_case_cuda (__main__.TestDropoutNNDeviceTypeCUDA.test_native_dropout_corner_case_cuda) ... ok

----------------------------------------------------------------------
Ran 15 tests in 0.541s

OK (skipped=1)

CC: @fffrog @albanD @jbschlosser

### CODE REVIEW

File: test/nn/test_dropout.py [hunk 1]
```diff
@@ -186,7 +159,7 @@ def test_Dropout(self, device):
 
         self._test_dropout_stride_mean_preserve(nn.Dropout, device)
 
-        if self.device_type == "cuda" or self.device_type == "cpu":
+        if self.device_type in ("cuda", "cpu", "xpu"):
             input = input.bfloat16()
             self._test_dropout(nn.Dropout, device, input)
```
Comment (fffrog) [inline test/nn/test_dropout.py]: Hey @kumartanmay-28, could you split this `test_Dropout` into two test cases? The first test case only focuses on the default dtype, and the second tests the `bfloat16` dtype, which is limited by the `cuda` and `cpu` accelerators.

File: test/nn/test_dropout.py [hunk 2]
```diff
@@ -322,6 +295,27 @@ def test_empty_dropout(self, device):
         out = torch.nn.functional.dropout(x)
         self.assertEqual(out.size(), x.size())
 
+    def test_native_dropout_corner_case(self, device):
+        # This test compares native_dropout on accelerator vs CPU, so skip when device is CPU
+        if self.device_type == "cpu":
+            self.skipTest("Test requires accelerator to compare with CPU")
```
Comment (fffrog) [inline test/nn/test_dropout.py]: Delete this one please

File: test/nn/test_dropout.py [hunk 3]
```diff
@@ -322,6 +295,27 @@ def test_empty_dropout(self, device):
         out = torch.nn.functional.dropout(x)
         self.assertEqual(out.size(), x.size())
 
+    def test_native_dropout_corner_case(self, device):
```
Comment (fffrog) [inline test/nn/test_dropout.py:302]: add `@onlyAccelerator` please

File: test/nn/test_dropout.py (hunk 1 above)
Comment (fffrog) [reply test/nn/test_dropout.py]: And `@dtypesIfCUDA` should be another good way :D

File: test/nn/test_dropout.py (hunk 3 above)
Comment (kumartanmay-28) [reply test/nn/test_dropout.py:302]: I'll make all the necessary changes! Thank you for the review @fffrog !

Comment (kumartanmay-28) [reply test/nn/test_dropout.py:302]: Hi @fffrog  All feedback addressed! Changes made:

  1. Split `test_Dropout` into two test cases:
     - `test_Dropout` - default dtype only
     - `test_Dropout_bfloat16` - bfloat16 with `@dtypesIfCUDA` and `@dtypes` decorators

  2. Removed CPU runtime skip from `test_native_dropout_corner_case`

  3. Added `@onlyAccelerator` decorator to `test_native_dropout_corner_case`

  Tests verified:
  - Default dropout on CPU: `test_Dropout_cpu` - OK
  - bfloat16 dropout on CUDA: `test_Dropout_bfloat16_cuda_bfloat16` - OK
  - Native dropout on CUDA: `test_native_dropout_corner_case_cuda` - OK
  - Native dropout skipped on CPU: `test_native_dropout_corner_case_cpu` - OK (skipped)

File: test/nn/test_dropout.py [hunk 4]
```diff
@@ -186,9 +161,11 @@ def test_Dropout(self, device):
 
         self._test_dropout_stride_mean_preserve(nn.Dropout, device)
 
-        if self.device_type == "cuda" or self.device_type == "cpu":
-            input = input.bfloat16()
-            self._test_dropout(nn.Dropout, device, input)
+    @dtypesIfCUDA(torch.bfloat16)
+    @dtypes(torch.bfloat16)
```
Comment (fffrog) [inline test/nn/test_dropout.py]:
```suggestion
    @dtypes(torch.bfloat16)
    @onlyOn(["cpu", "cuda"])
```

**Review by fffrog — APPROVED** [review]
Great work, thank you.

Just a tiny nit :D

**Comment by kumartanmay-28** [comment]
> Great work, thank you.
> 
> Just a tiny nit :D

Will update! Thanks @fffrog

File: test/nn/test_dropout.py [hunk 5]
```diff
@@ -186,9 +161,11 @@ def test_Dropout(self, device):
 
         self._test_dropout_stride_mean_preserve(nn.Dropout, device)
 
-        if self.device_type == "cuda" or self.device_type == "cpu":
-            input = input.bfloat16()
-            self._test_dropout(nn.Dropout, device, input)
+    @dtypes(torch.bfloat16)
+    @onlyOn(["cpu", "cuda"])
```
Comment (jbschlosser) [inline test/nn/test_dropout.py]: should this be `@onlyAccelerator` (with appropriate skips for unsupported functionality)?

**Review by jbschlosser — COMMENTED** [review]
awesome thank you :)

File: test/nn/test_dropout.py (hunk 5 above)
Comment (kumartanmay-28) [reply test/nn/test_dropout.py]: Hi @jbschlosser , Good catch! Changed to @onlyAccelerator. Both XPU and MPS support bfloat16. The original CPU/CUDA only restriction appears outdated. No skips needed IMO, bfloat16 dropout should work on all accelerators. Thank you! :)

**Review by fffrog — APPROVED** [review]
Sorry for the delay, SGTM, thank you.

**Review by albanD — APPROVED** [review]
